frontend/src/components/auth/Login.tsx

```diff
@@ -25,17 +40,24 @@ const Login = (): React.ReactElement => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [isSecondPage, setIsSecondPage] = useState<boolean>(false);
```

Comment:
I think we need (1) more descriptive names for this and (2) more than just two states, so maybe a type would work better? Something like

```ts
const [loginState, setLoginState] = useState<LoginState>(LoginState.EnterEmail);

interface LoginState {
  EnterEmail,
  EnterPassword,
  ForgotPassword,
};
```

and then getters to return the appropriate details? As in

```ts
const getBottomPrompt = (loginState: LoginState): React.ReactElement => {
  switch(loginState) {
    case LoginState.EnterEmail:
      return(<Text>Don't have an account? <Link>Sign up</Link></Text>);
    case:
      ...

  }
}
```

(I just typed this out of VScode so not sure it compiles, but hopefully the idea is somewhat clear)

Reply:
(Similar logic would also replace all the `isSecondPage ?` conditionals in the return func.) 

nit: It's also possible that we extract all each "return" clause into its own component... while that means more repeated code, it might also be more readable than having this much branching. I don't think there's too much downside either way, but maybe other people have suggestions? 

cc: @y2kwak @draymond63